hibernate-core/src/main/java/org/hibernate/persister/entity/JoinedSubclassEntityPersister.java

```diff
@@ -525,17 +526,36 @@ else if ( persistentClass.isDiscriminatorValueNotNull() ) {
 
 		int subclassSpan = persistentClass.getSubclassSpan() + 1;
 		subclassClosure = new String[subclassSpan];
-		subclassClosure[subclassSpan - 1] = getEntityName();
+		int subclassSpanMinusOne = subclassSpan - 1;
+		subclassClosure[subclassSpanMinusOne] = getEntityName();
 		if ( persistentClass.isPolymorphic() ) {
 			subclassesByDiscriminatorValue.put( discriminatorValue, getEntityName() );
 
-			discriminatorValuesByTableName = new LinkedHashMap<>( subclassSpan + 1 );
-			subclassNameByTableName = CollectionHelper.mapOfSize( subclassSpan + 1);
+			int initialCapacity = subclassSpan + 1;
+			discriminatorValuesByTableName = new LinkedHashMap<>( initialCapacity );
+			discriminatorColumnNameByTableName = new LinkedHashMap<>( initialCapacity );
```

Comment:
It seems `LinkedHashMap` shares the same problem with `HashMap` in that the `initialCapacity` is not the same as `expectedSize` except the loadFactor is specified as 1. Only `ConcurrentHashMap` is immune with this annoying problem. If `CollectionHelper` doesn't support util static factory method for it, we might well create one there.